src/java/voldemort/store/metadata/MetadataStore.java

```diff
@@ -531,31 +531,34 @@ public Object getCapability(StoreCapabilityType capability) {
             String key = ByteUtils.getString(keyBytes.get(), "UTF-8");
 
             if(METADATA_KEYS.contains(key) || this.storeNames.contains(key)) {
-                List<Versioned<byte[]>> values = Lists.newArrayList();
+                try {
+                    List<Versioned<byte[]>> values = Lists.newArrayList();
 
-                // Get the cached value and convert to string
-                Versioned<String> value = convertObjectToString(key, metadataCache.get(key));
+                    // Get the cached value and convert to string
+                    Versioned<String> value = convertObjectToString(key, metadataCache.get(key));
 
-                // Metadata debugging information
-                if(logger.isTraceEnabled())
-                    logger.trace("Key " + key + " requested, returning: " + value.getValue());
+                    // Metadata debugging information
+                    if(logger.isTraceEnabled())
+                        logger.trace("Key " + key + " requested, returning: " + value.getValue());
 
-                values.add(new Versioned<byte[]>(ByteUtils.getBytes(value.getValue(), "UTF-8"),
-                                                 value.getVersion()));
+                    values.add(new Versioned<byte[]>(ByteUtils.getBytes(value.getValue(), "UTF-8"),
+                                                     value.getVersion()));
 
-                return values;
+                    return values;
+                } catch(Exception e) {
+                    throw new VoldemortException("Failed to read metadata key:"
+                                                         + ByteUtils.getString(keyBytes.get(),
+                                                                               "UTF-8")
+                                                         + " delete config/.temp config/.version directories and restart.",
+                                                 e);
+                }
             } else {
-                throw new VoldemortException("Unhandled Key:" + key + " for MetadataStore get()");
+                throw new VoldemortException("Store " + key + " does not exist on node "
```

Comment:
Is this code only used for "stores" or are there other types of metadata that use the same code? If it's the latter, the more specific message may not make sense in all contexts, and it may be preferable to override the message higher up the call stack.

I don't know, I haven't looked at that code in detail.